packages/eas-cli/src/commands/metadata/pull.ts

```diff
@@ -27,18 +26,20 @@ export default class MetadataPull extends EasCommand {
   };
 
   static override contextDefinition = {
-    ...EASCommandProjectIdContext,
+    ...EASCommandDynamicProjectConfigContext,
     ...EASCommandLoggedInContext,
   };
 
   async runAsync(): Promise<void> {
     Log.warn('EAS Metadata is in beta and subject to breaking changes.');
 
     const { flags } = await this.parse(MetadataPull);
-    const { actor } = await this.getContextAsync(MetadataPull, { nonInteractive: false });
+    const { actor, getDynamicProjectConfigAsync } = await this.getContextAsync(MetadataPull, {
+      nonInteractive: false,
+    });
 
     const projectDir = await findProjectRootAsync();
-    const { exp } = getConfig(projectDir, { skipSDKVersionRequirement: true });
+    const { exp } = await getDynamicProjectConfigAsync({ skipSDKVersionRequirement: true });
```

Comment:
this is a bug, `getConfig` should not be used directly and when you use getExpoConfig  skipSDKVersionRequirement is passed there by default. You can use exp from EASCommandProjectConfigContext